runtime/bigarray.c

```diff
@@ -132,7 +132,7 @@ static uint16 caml_float_to_float16(float d)
       uint32_t mant_odd = (f.i >> 13) & 1; // resulting mantissa is odd
 
       // update exponent, rounding bias part 1
-      f.i += ((15 - 127) << 23) + 0xfff;
+      f.i += ((uint32_t)(15 - 127) << 23) + 0xfff;
```

Comment:
Do I understand well that this was undefined behaviour? I think this change deserves its own PR to make sure it is not lost in the shuffle. Thanks!

Reply:
Ok, but this branch won't build anymore if I remove this commit.

Reply:
Leave it in this PR if you wish, but a separate PR with this change would still be useful, I think, and would be quickly merged anyway.

Reply:
Done in #13094.